From 351f354b18a872e5911343ff1c95494892a572a7 Mon Sep 17 00:00:00 2001
From: Ger Hobbelt <ger@hobbelt.com>
Date: Wed, 22 Nov 2023 11:53:35 +0100
Subject: [PATCH] unify the code to default to using openCL 1.2 when no version
 has been specified. We use OpenCL 1.2 as the default as that SHOULD work okay
 on non-Intel+NVIDIA hardware (AMD, for example) and we want a multi-platform
 compatible build by default.

---
 CL/cl_version.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/headers/CL/cl_version.h b/headers/CL/cl_version.h
index 3844938d..a7183111 100644
--- a/headers/CL/cl_version.h
+++ b/headers/CL/cl_version.h
@@ -19,8 +19,8 @@
 
 /* Detect which version to target */
 #if !defined(CL_TARGET_OPENCL_VERSION)
-#pragma message("cl_version.h: CL_TARGET_OPENCL_VERSION is not defined. Defaulting to 310 (OpenCL 3.1)")
-#define CL_TARGET_OPENCL_VERSION 310
+#pragma message("cl_version.h: CL_TARGET_OPENCL_VERSION is not defined. Defaulting to 120 (OpenCL 1.2)")
+#define CL_TARGET_OPENCL_VERSION 120
 #endif
 #if CL_TARGET_OPENCL_VERSION != 100 && \
     CL_TARGET_OPENCL_VERSION != 110 && \
@@ -29,9 +29,9 @@
     CL_TARGET_OPENCL_VERSION != 220 && \
     CL_TARGET_OPENCL_VERSION != 300 && \
     CL_TARGET_OPENCL_VERSION != 310
-#pragma message("cl_version: CL_TARGET_OPENCL_VERSION is not a valid value (100, 110, 120, 200, 210, 220, 300, 310). Defaulting to 310 (OpenCL 3.1)")
+#pragma message("cl_version: CL_TARGET_OPENCL_VERSION is not a valid value (100, 110, 120, 200, 210, 220, 300). Defaulting to 120 (OpenCL 1.2)")
 #undef CL_TARGET_OPENCL_VERSION
-#define CL_TARGET_OPENCL_VERSION 310
+#define CL_TARGET_OPENCL_VERSION 120
 #endif
 
 
